src/sdl/YamlSDLParser.ts

```diff
@@ -0,0 +1,192 @@
+import { z } from "zod";
+import { BetaSdl, NetworkBeta2, NetworkBeta3, NetworkVersion } from "./types";
+import YAML from "js-yaml";
+
+export const v2HTTPOptions = z.object({
```

Comment:
this schemas and the ones below are exactly the same as the types defined in the `./types.ts`. Those types are now inferred from schemas as a primary source of trueth.